test/e2e/kube-ovn/underlay/underlay.go

```diff
@@ -417,4 +435,307 @@ var _ = framework.Describe("[group:underlay]", func() {
+					underlayCidr = append(underlayCidr, config.Subnet)
+					gateway = append(gateway, config.Gateway)
+				}
+			case apiv1.ProtocolIPv6:
+				if f.ClusterIpFamily != "ipv4" {
+					underlayCidr = append(underlayCidr, config.Subnet)
+					gateway = append(gateway, config.Gateway)
+				}
+			}
+		}
+
+		excludeIPs := make([]string, 0, len(network.Containers)*2)
+		for _, container := range network.Containers {
+			if container.IPv4Address != "" && f.ClusterIpFamily != "ipv6" {
+				excludeIPs = append(excludeIPs, strings.Split(container.IPv4Address, "/")[0])
+			}
+			if container.IPv6Address != "" && f.ClusterIpFamily != "ipv4" {
+				excludeIPs = append(excludeIPs, strings.Split(container.IPv6Address, "/")[0])
+			}
+		}
+
+		subnet := framework.MakeSubnet(subnetName, vlanName, strings.Join(underlayCidr, ","), strings.Join(gateway, ","), excludeIPs, nil, []string{namespaceName})
+		subnet.Spec.U2OInterconnection = true
+		_ = subnetClient.CreateSync(subnet)
+		ginkgo.By("Creating underlay subnet pod")
+		annotations := map[string]string{
+			util.LogicalSwitchAnnotation: subnetName,
+		}
+
+		u2oPodNameUnderlay = "pod-" + framework.RandomSuffix()
+		args := []string{"netexec", "--http-port", strconv.Itoa(curlListenPort)}
+		underlayPod := framework.MakePod(namespaceName, u2oPodNameUnderlay, nil, annotations, framework.AgnhostImage, nil, args)
+		underlayPod.Spec.Containers[0].ImagePullPolicy = corev1.PullIfNotPresent
+
+		originUnderlayPod := underlayPod.DeepCopy()
+		underlayPod = podClient.CreateSync(underlayPod)
+
+		// get subnet again because ipam change
+		subnet = subnetClient.Get(subnetName)
+
+		ginkgo.By("Creating overlay subnet")
+		podClient = f.PodClient()
+		subnetClient = f.SubnetClient()
+		namespaceName = f.Namespace.Name
+		u2oOverlaySubnetName = "subnet-" + framework.RandomSuffix()
+		cidr := framework.RandomCIDR(f.ClusterIpFamily)
+
+		overlaySubnet := framework.MakeSubnet(u2oOverlaySubnetName, "", cidr, "", nil, nil, nil)
+		overlaySubnet = subnetClient.CreateSync(overlaySubnet)
+
+		ginkgo.By("Creating overlay subnet pod")
+		u2oPodNameOverlay = "pod-" + framework.RandomSuffix()
+		overlayAnnotations := map[string]string{
+			util.LogicalSwitchAnnotation: overlaySubnet.Name,
+		}
+		args = []string{"netexec", "--http-port", strconv.Itoa(curlListenPort)}
+		overlayPod := framework.MakePod(namespaceName, u2oPodNameOverlay, nil, overlayAnnotations, framework.AgnhostImage, nil, args)
+		overlayPod.Spec.Containers[0].ImagePullPolicy = corev1.PullIfNotPresent
+		overlayPod = podClient.CreateSync(overlayPod)
+
+		ginkgo.By("step1: Enable u2o check")
+		checkU2OItems(true, subnet, underlayPod, overlayPod)
+
+		ginkgo.By("step2: Disable u2o check")
+		podClient.DeleteSync(u2oPodNameUnderlay)
+
+		subnet = subnetClient.Get(subnetName)
+		modifiedSubnet := subnet.DeepCopy()
+		modifiedSubnet.Spec.U2OInterconnection = false
+		subnetClient.PatchSync(subnet, modifiedSubnet)
+		time.Sleep(5 * time.Second)
```

Comment:
Why sleep 5s here? To wait for the subnet to be ready?

Reply:
yes， if create a pod and subnet adding or updating at the same time, it will trigger IP count error sometimes. and This problem seems to have always existed

Reply:
We should wait until the subnet is ready instead of sleeping a period.

Reply:
> yes， if create a pod and subnet adding or updating at the same time, it will trigger IP count error sometimes. and This problem seems to have always existed

I think this is a bug. Sometimes a user creates subnet(s) and pod(s) using a single YAML file.

Reply:
sorry, I describe it wrong, it causes errors only in updating the subnet, but not creating the subnet.

because when you update the subnet, the subnet status is always ready.

Reply:
it may not work, status.u2oInterconnectionIP is changed after function reconcileU2OInterconnectionIP

and if there is a pod added after the time subnet executes reconcileU2OInterconnectionIP， the subnet usingIps count may not include this pod.

I originally used sleep to do a workaround, because once the u2o switch is switched, all pods have to be rebuilt